docs/en/sql-reference/data-types/newjson.md

```diff
@@ -89,6 +89,7 @@ SELECT (tuple(42 AS b) AS a, [1, 2, 3] AS c, 'Hello, World!' AS d)::JSON AS json
 Using CAST from `Map`:
 
 ```sql
+SET enable_variant_type=1, use_variant_as_common_type=1;
```

Comment:
Somewhat related question: Does it even make sense to support `enable_variant_type=0, enable_json_type=1`? I'd expect that enabling JSON would enable variant too

Reply:
JSON uses Variant type inside yes, but it uses it in special controlled scenarios with limited number of types inside. Variant as a separate type can be used with any types combination in any scenarios user can think of and in theory it can lead to something unexpected. So I would say enabling JSON type should not lead to automatic Variant enabling